claim_batch/services.py

```diff
@@ -517,27 +519,43 @@ def do_process_batch(audit_user_id, location_id, period, year):
             #    logger.info("no payment plan product found for product  %s for %s/%s", product_id, period, year)
             # 4 update the claim Total amounts if all Item and services got "valuated"
             claims = Claim.objects\
-                .filter(validity_from__lte = start_date)\
-                .filter(validity_from__gte = end_date)\
-                .filter(validity_to__isnull = True)\
-                .filter(claim_pocess_stamp_lte = end_date  )\
-                .filter(claim_pocess_stamp_gte = end_date  )\
-                .filter(Q(items__product_in = products) & Q(items__legacy_id__isnull = True) | Q(services__product_in = products) & Q(services__legacy_id__isnull = True))\
+                .filter(validity_from__lte=end_date)\
+                .filter(validity_from__gte=start_date)\
+                .filter(validity_to__isnull=True)\
+                .filter(process_stamp__lte=end_date)\
+                .filter(process_stamp__gte=start_date)\
+                .filter(Q(items__product__in=products) & Q(items__legacy_id__isnull=True) | Q(services__product__in=products) & Q(services__legacy_id__isnull=True))\
```

Comment:
we might not need the service and item list as they are part of the claims

Reply:
yes. I was also thinking about this. Good point

Reply:
ok. I assume this part of code will look like below:
`claims = Claim.objects\
                .filter(validity_from__lte=end_date)\
                .filter(validity_from__gte=start_date)\
                .filter(validity_to__isnull=True)\
                .filter(process_stamp__lte=end_date)\
                .filter(process_stamp__gte=start_date)\`

Reply:
Or do we need also this part:
`.prefetch_related(Prefetch('items', queryset=ClaimItem.objects.filter(legacy_id__isnull=True)))\
                .prefetch_related(Prefetch('services', queryset=ClaimService.objects.filter(legacy_id__isnull=True)))`
??